src/unwinder/i386/unwinder_i386.cpp

```diff
@@ -105,8 +105,19 @@ OOPStackUnwinderX86::VirtualUnwind(
     ENUM_CALLEE_SAVED_REGISTERS();
 #undef CALLEE_SAVED_REGISTER
 
-    ContextRecord->Esp = rd.SP - codeInfo.GetCodeManager()->GetStackParameterSize(&codeInfo);
-    ContextRecord->ResumeEsp = rd.SP;
+    SIZE_T paramSize = codeInfo.GetCodeManager()->GetStackParameterSize(&codeInfo);
+    SIZE_T paddingSize = 0;
+
+#ifdef UNIX_X86_ABI
+    // On UNIX_X86_ABI, function call may have stack alignment padding.
+    if (paramSize % 16 != 0)
```

Comment:
I don't think this would work in all cases. The padding doesn't depend just on the params size, but also on the esp before the caller performed the padding and argument pushing. This would work only if the esp after the prolog was always guaranteed to be aligned. Which I think is not the case. Or am I wrong?
I've seen @seanshpark mentioned before that he's working on enabling the FEATURE_FIXED_OUT_ARGS. Should we wait for it instead? It seems it would solve this problem.